File: 卡码/代码随想录/1/08-贪心算法/Js/02-中等/LeetCode_0738.js
//738. 单调递增的数字
const monotoneIncreasingDigits = function(n) {
	let nums=n.toString().split("");
	let start=nums.length;
	for(let i=nums.length-1;i>0;i--){
		if(nums[i-1]>nums[i]){
			nums[i-1]--;
			start=i;
		}
	}
	for(let i=start;i<nums.length;i++) nums[i]=9;
	return nums.join("");
};

File: 卡码/代码随想录/1/08-贪心算法/C++/02-中等/LeetCode_0738.cpp
//738. 单调递增的数字
#include <iostream>
#include <algorithm>
using namespace std;

class Solution {
public:
	int monotoneIncreasingDigits(int n) {
		string nums=to_string(n);
		int start=nums.length();
		for(int i=nums.length()-1;i>0;i--){
			if(nums[i-1]>nums[i]){
				nums[i-1]--;
				start=i;
			}
		}
		for(int i=start;i<nums.length();i++) nums[i]='9';
		return stoi(nums);
	}
};

int main(){
  system("pause");
  return 0;
}
